tests/test-class-wc-payments-token-service.php

```diff
@@ -197,17 +215,39 @@ public function test_woocommerce_payment_token_set_default_no_customer() {
 		$this->token_service->woocommerce_payment_token_set_default( 'pm_mock', $token );
 	}
 
-	public function test_woocommerce_get_customer_payment_tokens() {
-		$token = new WC_Payment_Token_CC();
-		$token->set_gateway_id( 'woocommerce_payments' );
-		$token->set_token( 'pm_mock0' );
-		$token->set_card_type( 'visa' );
-		$token->set_last4( '4242' );
-		$token->set_expiry_month( 1 );
-		$token->set_expiry_year( 2023 );
-		$token->set_user_id( 1 );
-		$token->set_default( true );
-		$token->save();
+	public function test_woocommerce_get_customer_payment_tokens_removes_unavailable_tokens() {
+		$token1            = WC_Helper_Token::create_token( 'pm_mock0' );
+		$unavailable_token = WC_Helper_Token::create_token( 'pm_mock1' );
+		$token2            = WC_Helper_Token::create_token( 'pm_mock2' );
+
+		$token1->add_meta_data( '_wcpay_customer_id', 'cus_12345' );
+		$token2->add_meta_data( '_wcpay_customer_id', 'cus_12345' );
+		$unavailable_token->add_meta_data( '_wcpay_customer_id', 'cus_67890' );
+
+		$tokens = [
```

Comment:
It looks like `$tokens` passed to the WC filter might come from another gateways too. Let's add other gateway token to make sure it's not filtered out by our pipeline.

Reply:
Good catch, added in https://github.com/Automattic/woocommerce-payments/pull/973/commits/0ff2e7135db57e1cc80c4e6f02d5d26dd216efc8 and https://github.com/Automattic/woocommerce-payments/pull/973/commits/a625b21f42fb1799af0eef27626f5dbb7e4eab91.